scripts/bundle-image-pickup.sh

```diff
@@ -0,0 +1,182 @@
+    "\nPOSTGRES:\t\t${POSTGRES_IMAGE:-$DEFAULT_POSTGRES_IMAGE}" \
+    "\nSEARCH_API:\t\t${API_IMAGE:-$DEFAULT_API_IMAGE}" \
+    "\nSEARCH_COLLECTOR:\t${COLLECTER_IMAGE:-$DEFAULT_COLLECTOR_IMAGE}" \
+    "\nSEARCH_INDEXER:\t\t${INDEXER_IMAGE:-$DEFAULT_INDEXER_IMAGE}" \
+    "\nSEARCH_OPERATOR:\t${OPERATOR_IMAGE:-$DEFAULT_OPERATOR_IMAGE}" \
+    "\n==============================================================================\n"
+}
+
+get_default_images_from_csv () {
+  log_color purple "Fetching default component images from ${OPERATOR_CSV_FILEPATH}\n"
+
+  for IMG in $(yq e $OPERATOR_IMAGE_PATH $OPERATOR_CSV_FILEPATH); do
+    if [[ $IMG =~ .*"search-v2-operator".* ]]; then
+      DEFAULT_OPERATOR_IMAGE=$IMG
+    fi
+  done
+
+  for IMG in $(yq e $OPERATOR_ENV_PATH $OPERATOR_CSV_FILEPATH); do
+    if [[ $IMG =~ .*"postgres".* ]]; then
+      DEFAULT_POSTGRES_IMAGE=$IMG
+
+    elif [[ $IMG =~ .*"search-collector".* ]]; then
+      DEFAULT_COLLECTOR_IMAGE=$IMG
+
+    elif [[ $IMG =~ .*"search-indexer".* ]]; then
+      DEFAULT_INDEXER_IMAGE=$IMG
+
+    elif [[ $IMG =~ .*"search-v2-api".* ]]; then
+      DEFAULT_API_IMAGE=$IMG
+    fi
+  done
+
+  display_component_images
+}
+
+update_images_csv () {
+  COMPONENT=$1
+  NEW_IMAGE=$2
+  log_color purple "Preparing to update component: $COMPONENT => $NEW_IMAGE\n"
+
+  # TODO: Replace yq path with $OPERATOR_ENV_PATH. (Note: Adding the env variable seems to cause yq to return no results)
+  if [[ $COMPONENT =~ .*"search-indexer".* ]]; then
+    yq -i e '.spec.install.spec.deployments[0].spec.template.spec.containers[1].env[1].value = "'$NEW_IMAGE'"' $OPERATOR_CSV_FILEPATH
+
+  elif [[ $COMPONENT =~ .*"search-collector".* ]]; then
+    yq -i e '.spec.install.spec.deployments[0].spec.template.spec.containers[1].env[2].value = "'$NEW_IMAGE'"' $OPERATOR_CSV_FILEPATH
+
+  elif [[ $COMPONENT =~ .*"search-v2-api".* ]]; then
+    yq -i e '.spec.install.spec.deployments[0].spec.template.spec.containers[1].env[3].value = "'$NEW_IMAGE'"' $OPERATOR_CSV_FILEPATH
+    
+  # TODO: Replace yq path with $OPERATOR_IMAGE_PATH. (Note: Adding the env variable seems to cause yq to return no results)
+  elif [[ $COMPONENT =~ .*"search-v2-operator".* ]]; then
+    yq -i e '.spec.install.spec.deployments[0].spec.template.spec.containers[1].image = "'$NEW_IMAGE'"' $OPERATOR_CSV_FILEPATH
+  fi 
+}
+
+log_color "cyan" "Initializing search bundle image pickup..."
+echo -e "Current dir: $(pwd)\n"
+
+# Create an array containing the Search components that we will focus on for image versioning
+SEARCH_COMPONENTS=(postgresql-13 search-collector search-indexer search-v2-api search-v2-operator)
+get_default_images_from_csv
+
+# Clone the pipeline repository (We need to fetch the latest manifest file to capture the latest builds)
+curl -o $PIPELINE_MANIFEST_FILEPATH https://raw.githubusercontent.com/$ORG/$PIPELINE_REPO/$RELEASE_BRANCH/manifest.json -H "Authorization: token $GITHUB_TOKEN"
```

Comment:
You could save it to a variable so we don't have to worry about cleaning the file.
```bash
$PIPELINE_MANIFEST=$(curl -o $PIPELINE_MANIFEST_FILEPATH https://raw.githubusercontent.com/$ORG/$PIPELINE_REPO/$RELEASE_BRANCH/manifest.json -H "Authorization: token $GITHUB_TOKEN")
```

Then below use it like this.
```bash
MANIFEST_JSON=$(echo $PIPELINE_MANIFEST | jq '.[] | select(."image-name" | match("'$COMPONENT'";"i"))')
```

Reply:
@dislbenn Can we log the entire file before the update and after the update ? Will be helpful to test.

Reply:
We can create a markdown file and list the component versions before and after the script is executed? Every time the script is executed, it can add a new entry to the doc. Is this the kind of logging you are referring to?